### PR TITLE
Add `imported` field to private keys

### PR DESCRIPTION
**Pull request by samholmes**
---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1201810828119037

### CODE REVIEW

File: docs/key-formats.md [hunk 1]
```diff
@@ -83,6 +84,8 @@ If `format` is missing, the wallet uses a default of bip32.
 
 If the `coinType` is missing, the default is 0 for Bitcoin. Other coins use different defaults for `coinType`.
 
+The `imported` field signifies whether the wallet was imported or created. If the `imported` field is `undefined`, then the wallet must have been created before this field was added to the specification and whether the wallet was imported or created is unknown. Implementations should specify `true` if the wallet was imported, and `false` if the wallet was created moving forward.
+
```
Comment (swansontec) [inline docs/key-formats.md:88]: I would word this differently:

> The `imported` field should be `true` for keys that the user imported, and `false` for keys that the app generated randomly. The `imported` field is optional, and if it is missing, the keys were created before the app began recording this information.

Also, I would move this paragraph to the "Storage keys" section, which should then be renamed to "Common key properties" or such.

File: src/core/account/plugin-api.js [hunk 2]
```diff
@@ -69,7 +69,9 @@ export class CurrencyConfig extends Bridgeable<EdgeCurrencyConfig> {
     if (tools.importPrivateKey == null) {
       throw new Error('This wallet does not support importing keys')
     }
-    return tools.importPrivateKey(userInput, opts.keyOptions)
+    const keys = await tools.importPrivateKey(userInput, opts.keyOptions)
+    keys.imported = true
+    return keys
```
Comment (swansontec) [inline src/core/account/plugin-api.js:74]: Can we change this structure to `return { ...keys, imported: true }`. That will keep TypeScript happy after we run the conversion script.

File: src/core/login/keys.js [hunk 3]
```diff
@@ -319,8 +319,10 @@ export async function createCurrencyWallet(
       throw new Error('This wallet does not support importing keys')
     }
     keys = await tools.importPrivateKey(opts.importText, opts.keyOptions)
+    keys.imported = true
```
Comment (swansontec) [inline src/core/login/keys.js:322]: `keys = { ...await tools.importPrivateKey(), imported: true }`

File: src/core/login/keys.js [hunk 4]
```diff
@@ -319,8 +319,10 @@ export async function createCurrencyWallet(
...
   } else {
     keys = await tools.createPrivateKey(walletType, opts.keyOptions)
+    keys.imported = false
```
Comment (swansontec) [inline src/core/login/keys.js:325]: `keys = { ...await tools.createPrivateKey(), imported: false }`

**Review by swansontec — CHANGES_REQUESTED** [review]

**Review by swansontec — APPROVED** [review]

File: src/core/login/keys.js [hunk 5]
```diff
@@ -318,9 +318,15 @@ export async function createCurrencyWallet(
...
-    keys = await tools.importPrivateKey(opts.importText, opts.keyOptions)
+    keys = {
+      ...(await tools.importPrivateKey(opts.importText, opts.keyOptions)),
+      imported: true
+    }
   } else {
-    keys = await tools.createPrivateKey(walletType, opts.keyOptions)
+    keys = {
+      ...(await tools.createPrivateKey(walletType, opts.keyOptions)),
+      imported: true
```
Comment (swansontec) [inline src/core/login/keys.js:328]: This one needs to be false.

**Review by swansontec — CHANGES_REQUESTED** [review]
You got the flag backwards in your fixup.

File: src/core/login/keys.js (hunk 5 above)
Comment (samholmes) [reply src/core/login/keys.js:328]: Oops. That was a derp on my part. Fixed up.

**Review by swansontec — APPROVED** [review]

**Pull request merged**